internal/rsl/rsl.go

```diff
@@ -152,6 +161,38 @@ func (a Annotation) createCommitMessage() (string, error) {
 	return strings.Join(lines, "\n"), nil
 }
 
+func GetFirstEntry(repo *git.Repository) (EntryType, error) {
+	ref, err := repo.Reference(plumbing.ReferenceName(RSLRef), true)
+	if err != nil {
+		return nil, err
+	}
+
+	iteratorHash := ref.Hash()
+	for {
+		commitObj, err := repo.CommitObject(iteratorHash)
+		if err != nil {
+			return nil, ErrRSLEntryNotFound
+		}
+
+		if len(commitObj.ParentHashes) == 0 {
+			// Found first entry
+			tmpEntry, err := parseRSLEntryText(commitObj.Hash, commitObj.Message)
+			if err != nil {
+				return nil, err
+			}
+
+			entry := tmpEntry.(*Entry)
```

Comment:
Will this ever not be a `*Entry`?

If yes, consider changing `parseRSLEntryText` to return a `*Entry`.
If no, add a type check to handle the case if a non-`*Entry` is returned.

Reply:
Good point, the first entry is expected to be `*Entry` and not `*Annotation` because it's always for the policy. I'll change the return type and pull out the RSL specific changes as a separate PR